ports/stm32/accel.c

```diff
@@ -226,3 +232,180 @@ const mp_obj_type_t pyb_accel_type = {
 };
 
 #endif // MICROPY_HW_HAS_MMA7660
+
+
+#if MICROPY_HW_HAS_KXTJ3
+
+/// \moduleref pyb
+/// \class Accel - accelerometer control
+///
+/// Accel is an object that controls the KXTJ3 accelerometer.
+//  Example usage:
+///
+///     accel = pyb.Accel()
+///     for i in range(10):
+///         print(accel.x(), accel.y(), accel.z())
+///
+/// Raw values are between -128 and 127 for -/+ 2G acceleration.
+
+
+#define I2C_TIMEOUT_MS (50)
+
+#define KXT_ADDR            (0x0F)
+#define KXT_REG_DCST_RESP   (0x0C)
+#define KXT_REG_WHO_AM_I    (0x0F)
+#define KXT_REG_XOUT_H      (0x07)
+#define KXT_REG_YOUT_H      (0x09)
+#define KXT_REG_ZOUT_H      (0x0B)
+#define KXT_REG_CTRL_REG1   (0x1B)
+#define KXT_REG_CTRL_REG2   (0x1D)
```

Comment:
If you rename `KXT_ADDR` -> `MMA_ADDR`, `KXT_REG_XOUT_H` -> `MMA_REG_X`, etc, then you can reuse a lot of the existing functions (like `read_axis`, `pyb_accel_x` etc).